azkaban-web-server/src/main/java/azkaban/flowtrigger/quartz/FlowTriggerScheduler.java

```diff
@@ -124,15 +126,13 @@ public void scheduleAll(final Project project, final String submitUser)
           final JobDataMap jobDataMap = job.getJobDataMap();
 
           final String flowId = jobDataMap.getString(FlowTriggerQuartzJob.FLOW_ID);
-
-          final String projectJson = jobDataMap.getString(FlowTriggerQuartzJob.PROJECT);
-          final Project project = FlowUtils.toProject(projectJson);
+          final int projectId = jobDataMap.getInt(FlowTriggerQuartzJob.PROJECT_ID);
           final FlowTrigger flowTrigger = (FlowTrigger) jobDataMap
               .get(FlowTriggerQuartzJob.FLOW_TRIGGER);
           final String submitUser = jobDataMap.getString(FlowTriggerQuartzJob.SUBMIT_USER);
           final List<? extends Trigger> quartzTriggers = quartzScheduler.getTriggersOfJob(jobKey);
           scheduledFlowTrigger = new ScheduledFlowTrigger(
```

Comment:
Interested to learn why does ScheduledFlowTrigger object need project name? If not, you don't need to inject project object, I believe.

Reply:
it's for UI display.